### PR TITLE
Change Ipopt to IDAES' Ipopt

### PR DESCRIPTION
**Pull request by mrmundt**
In order to address [Pyomo PR 1363](https://github.com/Pyomo/pyomo/pull/1363), Ipopt needs to be updated to IDAES' Ipopt.

(**NOTE**: IDAES does not have a version for Ubuntu 16, which is what the Travis jobs run on. So I'm not sure if it will work.)

### CODE REVIEW

**Comment by mrmundt** [comment]
@ghackebeil - I have lots of issues with "Permission denied" errors. Do you think maybe you need to submit the changes? Or did I do something to a file that triggered something permission-based?

**Comment by ghackebeil** [comment]
`mkdir: cannot create directory ‘/root/IDAES_2.0.0/IDAES-IPOPT-2.0.0’: No such file or directory`

I think maybe you just need to use the `-p` option with mkdir.

**Comment by mrmundt** [comment]
Interesting. So why does it work without issue when you run `mkdir <whatever>` in some of the other scripts (i.e., `gams.sh`)?

**Comment by ghackebeil** [comment]
The -p option is needed if there are intermediate directories in the path that do not already exist.

**Comment by mrmundt** [comment]
Fair enough, thanks! It looks like it's working this time.

**Comment by ghackebeil** [comment]
Nice

**Review by ghackebeil — APPROVED** [review]

File: linux_install_scripts/ipopt.sh [hunk 1]
```diff
@@ -3,27 +3,13 @@ RUN echo "" && \
...
-    mkdir build && \
-    cd build && \
-    ../configure CXX=g++ CC=gcc F77=gfortran ADD_CPPFLAGS="-DENABLE_RUN_MPC" > /dev/null && \
-    make -j$(nproc) > /dev/null && \
-    make install > /dev/null && \
-    cd Ipopt/contrib/sIPOPT && \
-    make -j$(nproc) > /dev/null && \
-    make install > /dev/null
+ARG TARGET="idaes-solvers-ubuntu1804-64"
+ENV IPOPT_VERSION="2.0.0"
```
Comment (jsiirola) [inline linux_install_scripts/ipopt.sh]: We should probably not mark this as the "IPOPT_VERSION", but rather the "IDAES_EXT_VERSION".  This is still distributing Ipopt 3.x

File: linux_install_scripts/ipopt.sh [hunk 2]
```diff
@@ -3,27 +3,13 @@ RUN echo "" && \
...
-    cd Lapack && ./get.Lapack 2> /dev/null && cd .. && \
-    cd Metis && ./get.Metis 2> /dev/null && cd .. && \
-    cd Mumps && ./get.Mumps 2> /dev/null && cd .. && \
-    cd .. && \
-    mkdir build && \
-    cd build && \
-    ../configure CXX=g++ CC=gcc F77=gfortran ADD_CPPFLAGS="-DENABLE_RUN_MPC" > /dev/null && \
-    make -j$(nproc) > /dev/null && \
-    make install > /dev/null && \
-    cd Ipopt/contrib/sIPOPT && \
```
Comment (jsiirola) [inline linux_install_scripts/ipopt.sh:26]: Note that by removing our build of Ipopt, we are losing `sipopt`

**Comment by jsiirola** [comment]
OK - I added a simple "attempt to run ipopt", and it fails due to a missing libgfortran4.  I believe that 16.04 only ships libgfortran3.  We may be stuck unless IDAES will build for 16.04, or we upgrade these images to 18.04.

@ghackebeil, is there a reason we are using 16.04 and not 18.04?

**Review by jsiirola — DISMISSED** [review]
This doesn't appear to result in a working ipopt installation.

**Comment by ghackebeil** [comment]
No particular reason. If you can find a different base image then feel free to change it.

**Comment by mrmundt** [comment]
@jsiirola @ghackebeil - Can someone please grant me access to https://hub.docker.com/r/pyomo/linux-test-images? It looks like that's where our base image lives, and if we want to update Ubuntu, I likely need to modify that, correct?

**Comment by ghackebeil** [comment]
There are login credentials encrypted into the Travis config file for this repo. I can share those with you, but I don’t think you would need them as everything about the images is configured within this repo. Once you push changes to master, if the Travis builds succeed, they will be pushed to the dockerhub site.

Make sense?

**Comment by ghackebeil** [comment]
Also the base image for each Python version is defined by the SRC_IMAGE variable in the travis.yml build matrix.

I’m basing ours off of the official Python, PyPy, and Anaconda provided images. You might have some flexibility in choosing what base images they are built off of, but maybe not what you want exactly. You’ll have to visit their dockerhub pages to see what is available.

**Comment by mrmundt** [comment]
> Also the base image for each Python version is defined by the SRC_IMAGE variable in the travis.yml build matrix.
> 
> I’m basing ours off of the official Python, PyPy, and Anaconda provided images. You might have some flexibility in choosing what base images they are built off of, but maybe not what you want exactly. You’ll have to visit their dockerhub pages to see what is available.

Maybe you can help me with one question - for things like Python 2.7, it lists a bunch of different OS'. How do you designate which OS you want it to use?

**Comment by ghackebeil** [comment]
Honestly, it was trial and error. Sometimes the default distro was too old and another package wouldn't build.

**Comment by ghackebeil** [comment]
Sorry I don’t think I answered the question you asked, but the Travis file might have some examples that help figure that out.

File: linux_install_scripts/ipopt.sh [hunk 3]
```diff
@@ -12,5 +12,5 @@ RUN mkdir -p ${PREFIX}/IDAES_${IDAES_VERSION}/${IPOPT_DIR} && \
     wget -q "https://github.com/IDAES/idaes-ext/releases/download/2.0.0/idaes-solvers-ubuntu1804-64.tar.gz" && \
     tar -xzf ${TARGET}.tar.gz && \
     rm -rf ${TARGET}.tar.gz
-RUN ${PREFIX}/IDAES_${IDAES_VERSION}/${IPOPT_DIR}/ipopt
+RUN ${PREFIX}/IDAES_${IDAES_VERSION}/${IPOPT_DIR}/ipopt -h
```
Comment (blnicho) [inline linux_install_scripts/ipopt.sh]: @mrmundt try `--help` or `--version`

**Comment by mrmundt** [comment]
@jsiirola @blnicho - It has passed all checks now!

**Review by blnicho — APPROVED** [review]

**Review by jsiirola — APPROVED** [review]
I think this looks fine.

**Pull request merged**